### DISCUSSION TITLE
Prefilling the create new file / edit file form

### DISCUSSION THREAD

**Discussion by Justineo** [community/community#8106] [Other Feature Feedback, Questions, & Ideas]
First of all, thank you for the continuous improvements to GitHub.

We can now create/edit files on GitHub and that's a great feature which makes it a lot more easier to make one-time contributions. My major use case is that I'm creating an interactive example editor for my frontend component library's docs. We provide a live code editor and let users play around and make modifications to the example code to see results. And it occurs to me what if users can improve example code directly in our code editor and if they made improvements to the example they can directly contribute back to repos on GitHub with ease.

![image](https://user-images.githubusercontent.com/1726061/143435399-3fd8087a-56c6-4dbf-acad-f77fef07187e.png)

After some investigation I found a post on reddit which provides some insight: https://www.reddit.com/r/github/comments/3ty3g9/github_trick_how_to_prefill_create_new_file_form/

It seems that some fields like `filename`, `value`, `message`, `description` can be prefilled but it doesn't seem to update the state of the submit button, and the `value` field doesn't seem to work when editing exisiting files. Also note that file contents can be long so `GET` requests may not be able to reliably carry the content over.

So my proposal here is that we make the prefilling feature *official*, like those supported for online playgrounds like CodePen, CodeSandbox and Stackblitz.

**Comment by airbreather** [comment]
See also: isaacs/github#1527

Edit: the issues described on that page are no longer occurring today (hooray!); I'm just linking to that issue to show a few more thumbs-ups re: the ask for a fully supported version of this feature.
